os/fs/smartfs/smartfs_utils.c

```diff
@@ -1276,33 +1291,24 @@ int smartfs_writeentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s new_
 	entrysize = sizeof(struct smartfs_entry_header_s) + fs->fs_llformat.namesize;
 	psector = new_entry.dsector;
 	offset = new_entry.doffset;
+	last_parentdirsector = new_entry.last_dsector;
 
 	/* If we are working with an invalidated/empty entry, the flags need to be set for the new entry. */
-	if (IS_AVAILABLE_ENTRY(&new_entry)) {
-		/* If it is an invalidated entry, flags need to be reset first. */
-#ifdef CONFIG_SMARTFS_ALIGNED_ACCESS
-		if (smartfs_rdle16(&new_entry.flags) != SMARTFS_ERASEDSTATE_16BIT) {
-#else
-		if (new_entry.flags != SMARTFS_ERASEDSTATE_16BIT) {
-#endif
-			RESET_ENTRY_FLAGS(&new_entry);
-		}
-		SET_ENTRY_FLAGS(new_entry.flags, mode, type);
-	}
+	smartfs_set_entry_flags(new_entry, type, mode);
 
-	/* If passed parameter parentdirsector != new_entry.dsector, it means it is a new chain sector that we will write to */
-	if (parentdirsector != psector) {
+	/* If passed parameter new_entry.last_dsector != new_entry.dsector, it means it is a new chain sector that we will write to */
+	if (last_parentdirsector != psector) {
```

Comment:
if (dsector != last_dsector) is correct approach, am I right?

Reply:
Using aliases for both as is the general tradition in all functions